shell/platform/windows/flutter_windows_view.h

```diff
@@ -0,0 +1,224 @@
+                double delta_x,
+                double delta_y,
+                int scroll_offset_multiplier) override;
+
+  // |WindowBindingHandlerDelegate|
+  void OnFontChange() override;
+
+ private:
+  // Sends a window metrics update to the Flutter engine using current window
+  // dimensions in physical
+  void SendWindowMetrics(size_t width, size_t height, double dpiscale) const;
+
+  // Reports a mouse movement to Flutter engine.
+  void SendPointerMove(double x, double y);
+
+  // Reports mouse press to Flutter engine.
+  void SendPointerDown(double x, double y);
+
+  // Reports mouse release to Flutter engine.
+  void SendPointerUp(double x, double y);
+
+  // Reports mouse left the window client area.
+  //
+  // Win32 api doesn't have "mouse enter" event. Therefore, there is no
+  // SendPointerEnter method. A mouse enter event is tracked then the "move"
+  // event is called.
+  void SendPointerLeave();
+
+  // Reports a keyboard character to Flutter engine.
+  void SendText(const std::u16string&);
+
+  // Reports a raw keyboard message to Flutter engine.
+  void SendKey(int key, int scancode, int action, char32_t character);
+
+  // Reports scroll wheel events to Flutter engine.
+  void SendScroll(double x,
+                  double y,
+                  double delta_x,
+                  double delta_y,
+                  int scroll_offset_multiplier);
+
+  // Sets |event_data|'s phase to either kMove or kHover depending on the
+  // current primary mouse button state.
+  void SetEventPhaseFromCursorButtonState(
+      FlutterPointerEvent* event_data) const;
+
+  // Sends a pointer event to the Flutter engine based on given data.  Since
+  // all input messages are passed in physical pixel values, no translation is
+  // needed before passing on to engine.
+  void SendPointerEventWithData(const FlutterPointerEvent& event_data);
+
+  // Resets the mouse state to its default values.
+  void ResetMouseState() { mouse_state_ = MouseState(); }
+
+  // Updates the mouse state to whether the last event to Flutter had at least
+  // one mouse button pressed.
+  void SetMouseFlutterStateDown(bool is_down) {
+    mouse_state_.flutter_state_is_down = is_down;
+  }
+
+  // Updates the mouse state to whether the last event to Flutter was a kAdd
+  // event.
+  void SetMouseFlutterStateAdded(bool is_added) {
+    mouse_state_.flutter_state_is_added = is_added;
+  }
+
+  // Updates the currently pressed buttons.
+  void SetMouseButtons(uint64_t buttons) { mouse_state_.buttons = buttons; }
+
+  // Currently configured WindowsRenderTarget for this view used by
+  // surface_manager for creation of render surfaces and bound to the physical
+  // os window.
+  std::unique_ptr<WindowsRenderTarget> render_target_;
+
+  // An object used for intializing Angle and creating / destroying render
+  // surfaces. Surface creation functionality requires a valid render_target.
+  std::unique_ptr<AngleSurfaceManager> surface_manager_ = nullptr;
```

Comment:
Nit: No need to initialize unique_ptr to nullptr.

Reply:
Fixed in next commit